src/normalizer/mod.rs

```diff
@@ -78,17 +72,127 @@ pub trait Normalizer: Sync + Send {
     /// Normalize the provided [`Token`].
     /// Options can be set using the provided [`NormalizerOption`].
     ///
-    /// A Normalizer can return several `Token`s.
-    fn normalize<'o>(
-        &self,
-        token: Token<'o>,
-        options: NormalizerOption,
-    ) -> Box<dyn Iterator<Item = Token<'o>> + 'o>;
+    fn normalize<'o>(&self, token: Token<'o>, options: NormalizerOption) -> Token<'o>;
+
+    /// Return true if the normalizer can process Token of a specific [`Script`] and [`Language`].
+    ///
+    /// Some normalizer are specialized for a `Script` or/and a `Language` and shouldn't be called on every `Token`s.
```

Comment:
How does one write a normalizer where `should_normalize` is not called on every token?

Reply:
No every normalizer are not called for every token, but, should_normalize is always called to know if a normalizer should be called